compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -212,15 +212,20 @@ class TreeUnpickler(reader: TastyReader,
 
     /** Read names in an interleaved sequence of (parameter) names and types/bounds */
     def readParamNames(end: Addr): List[Name] =
```

Comment:
I think there's a better way to express this. Work with these two functions instead:
```scala
    /** Read names in an interleaved sequence of types/bounds and (parameter) names,
     *  possibly followed by a sequence of modifiers.
     */
    def readParamNamesAndMods(end: Addr): (List[Name], FlagSet) =
      val paramNames =
        collectWhile(currentAddr.index < end.index && !isModifierTag(nextByte)) {
          skipTree()
          readName()
        }
      var mods = EmptyFlags
      until(end) { 
        readByte() match
          case IMPLICIT => mods |= Implicit
          case ERASED   => mods |= Erased
          case GIVEN    => mods |= Given
      }
      (paramNames, mods)

    /** Read `n` parameter types or bounds which are interleaved with names */
    def readParamTypes([T <: Type](n: Int)(using Context): List[T] =
      if n == 0 then Nil
      else
        val t = readType()
        readNat() // skip the name
        t :: readParamTypes(n - 1)
```